nio-impl/src/main/java/org/xnio/nio/NioXnioWorker.java

```diff
@@ -114,35 +118,8 @@
             }
         }
         this.workerThreads = workerThreads;
-        mbeanHandle = NioXnio.register(new XnioWorkerMXBean() {
-            public String getProviderName() {
-                return "nio";
-            }
-
-            public String getName() {
-                return workerName;
-            }
-
-            public boolean isShutdownRequested() {
-                return isShutdown();
-            }
-
-            public int getCoreWorkerPoolSize() {
-                return NioXnioWorker.this.getCoreWorkerPoolSize();
-            }
-
-            public int getMaxWorkerPoolSize() {
-                return NioXnioWorker.this.getMaxWorkerPoolSize();
-            }
-
-            public int getIoThreadCount() {
-                return threadCount;
-            }
-
-            public int getWorkerQueueSize() {
-                return NioXnioWorker.this.getWorkerQueueSize();
-            }
-        });
+        this.metrics = new NioWorkerMatrics(workerName);
```

Comment:
Anyway we should just keep them as MXBean implementations to minimize API change.